students/jakevanhulzen/Lab1Part4_DesignChallengeFINAL/Lab1Part4_DesignChallengeFINAL.ino

```diff
@@ -0,0 +1,118 @@
+const int BLED = 9;
+const int GLED = 10;
+const int RLED = 11;
+const int BUTTON = 2;
+
+boolean lastButton = LOW;
+boolean currentButton = LOW;
+int ledMode = 0;
+unsigned long previousMillis = 0;
+const long interval = 100;
+
+// State variables for fading modes
+int purpleBrightness = 0;
+int fadeDirection = 5;  // Incremental fade
+unsigned long fadeInterval = 20;  // Interval between each step in fading
+
+void setup() {
+  pinMode(BLED, OUTPUT);
+  pinMode(GLED, OUTPUT);
+  pinMode(RLED, OUTPUT);
+  pinMode(BUTTON, INPUT);
+}
+
+boolean debounce(boolean last) {
+  boolean current = digitalRead(BUTTON); // Read the button state
+  if (last != current) {
+    delay(5); // Wait 5ms for debouncing
+    current = digitalRead(BUTTON);
+  }
+  return current;
+}
+
+void setMode(int mode) {
+  unsigned long currentMillis = millis();
+
+  // OFF
+  if (mode == 0) {
+    digitalWrite(RLED, LOW);
+    digitalWrite(GLED, LOW);
+    digitalWrite(BLED, LOW);
+  }
+  // WHITE BLINKING (GREEN and RED and BLUE)
+  else if (mode == 1) {
+    static boolean state = LOW;
+    if (currentMillis - previousMillis >= 1000) {
```

Comment:
Try to avoid hard-coded numbers like `1000`. Instead use a constant variable with a descriptive name - it makes it easier to read and easier to edit/debug.